pipeline/transformer/transformer_tei/src/main/scala/weco/pipeline/transformer/tei/transformers/TeiLanguages.scala

```diff
@@ -2,64 +2,53 @@ package weco.pipeline.transformer.tei.transformers
 
 import weco.catalogue.internal_model.languages.Language
 import weco.pipeline.transformer.result.Result
-import weco.pipeline.transformer.tei.TeiXml
 import weco.pipeline.transformer.tei.data.TeiLanguageData
+import cats.syntax.traverse._
+import cats.instances.either._
 
-import scala.util.{Failure, Success, Try}
-import scala.xml.Elem
+import scala.xml.{Elem, NodeSeq}
 
 object TeiLanguages {
 
+  def apply(xml: Elem): Result[List[Language]] =
+    parseLanguages(xml \\ "msDesc" \ "msContents")
+
   /** The languages of the TEI are in `textLang` nodes under `msContents`.
     *
     * <TEI xmlns="http://www.tei-c.org/ns/1.0" xml:id={id}>
-    *  <teiHeader>
-    *    <fileDesc>
-    *      <sourceDesc>
-    *        <msDesc xml:lang="en" xml:id="MS_Arabic_1">
-    *          <msContents>
-    *            <textLang mainLang={id} source="IANA">{label}</textLang>
+    * <teiHeader>
+    * <fileDesc>
+    * <sourceDesc>
+    * <msDesc xml:lang="en" xml:id="MS_Arabic_1">
+    * <msContents>
+    * <textLang mainLang={id} source="IANA">{label}</textLang>
```

Comment:
Something weird has happened to this formatting.

Reply:
damn you IntelliJ..